src/pull-request-core.ts

```diff
@@ -0,0 +1,234 @@
+import Octokit, { GitUpdateRefResponse } from '@octokit/rest';
+
+import {
+    PullRequest,
+    PullRequestWithChangesetCommit,
+    SourceFile
+} from './types';
+
+export function issuePullRequestWithToken(
+    { pullRequest, changesetCommit }: PullRequestWithChangesetCommit,
+    githubAccessToken: string
+): Promise<Octokit.PullsCreateResponse> {
+    const octokit = new Octokit({
+        auth: githubAccessToken
+    });
+
+    return Promise.resolve(issuePullRequest({ pullRequest, changesetCommit }));
```

Comment:
I coded things this way to provoke some feedback :P 

Below this line are the helper functions that fire off the PR chain. They use the `octokit` client as a dependency. I wrapped all of these functions inside this `issuePullRequestWithToken` so that the `octokit` client is in scope where I create it above.

 The alternatives I considered were: (a) thread the `octokit` client through each of the helper functions (don't like that much) or (b) creating a class with a constructor (don't like that much either).

What's the preferred TypeScript way to do this?

Reply:
This is fine and there is no strong convention. I'd prefer passing the client through a parameter (a consistently named property in an options bag) because that allows you to move functions into different files and also unit test them (which is needed if the functions get more complicated). See e.g. here: https://github.com/sourcegraph/create/blob/master/src/github.ts
But this is fine too since there is not that much code here.

Reply:
Gotcha. Yeah testability is important and some of this may grow and will need to be exposed.

Abstractly (i.e., given arbitrary language) if I have to choose between passing <thing> along vs. parameterizing <thing> over a set of tightly coupled functions (for number of functions >3, say, and where I don't anticipate moving functions into other files) I prefer the latter. Classes seem to be the obvious native TS way to achieve that? So even though I avoid classes in general, there's nothing mutable about this object, so it seems to work out OK.